### PR TITLE
Bump VMs, to Ubuntu 2204 with cgroups v1

### PR DESCRIPTION
**Pull request by edsantiago**
...and enable the at-test-time confirmation, the one that
double-checks that if CI requests runc we actually use runc.
This exposed a nasty surprise in our setup: there are steps to
define $OCI_RUNTIME, but that's actually a total fakeout!
OCI_RUNTIME is used only in e2e tests, it has no effect
whatsoever on actual podman itself as invoked via command
line such as in system tests. Solution: use containers.conf
 
Given how fragile all this runtime stuff is, I've also added
new tests (e2e and system) that will check $CI_DESIRED_RUNTIME.
 
Image source: https://github.com/containers/automation_images/pull/146
 
Since we haven't actually been testing with runc, we need
to fix a few tests:
 
  - handle an error-message change (make it work in both crun and runc)
  - skip one system test, "survive service stop", that doesn't
    work with runc and I don't think we care.
 
...and skip a bunch, filing issues for each:
 
  - #15013 pod create --share-parent
  - #15014 timeout in dd
  - #15015 checkpoint tests time out under $CONTAINER
  - #15017 networking timeout with registry
  - #15018 restore --pod gripes about missing --pod
  - #15025 run --uidmap broken
  - #15027 pod inspect cgrouppath broken
  - ...and a bunch more ("podman pause") that probably don't
    even merit filing an issue.
 
Also, use /dev/urandom in one test (was: /dev/random) because
the test is timing out and /dev/urandom does not block. (But
the test is still timing out anyway, even with this change)
 
Also, as part of the VM switch we are now using go 1.18 (up
from 1.17) and this broke the gitlab tests. Thanks to @Luap99
for a quick fix.
 
Also, slight tweak to #15021: include the timeout value, and
reword message so command string is at end.
 
Also, fixed a misspelling in a test name.
 
Fixes: #14833 

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
We are now testing with runc again. Golang has been updated from 17 to 18.
```

### CODE REVIEW

**Comment by adrianreber** [comment]
@rst0git knows how to update the obs and launchpad packages

**Comment by rst0git** [comment]
@edsantiago I pushed an update for version 3.17.1 in [OBS](https://build.opensuse.org/package/show/devel:tools:criu/criu).

**Comment by edsantiago** [comment]
@rst0git thank you! I see what looks like an [ubuntu 2204 log](https://build.opensuse.org/package/live_build_log/devel:tools:criu/criu/xUbuntu_22.04/x86_64) indicating that it succeeded (assuming 'xUbuntu' == 'Ubuntu'). Do you have a sense for how long that will then take to get into standard Ubuntu repos?

**Comment by rst0git** [comment]
> Do you have a sense for how long that will then take to get into standard Ubuntu repos?

A simple test with Ubuntu 22.04 container indicates that the criu package has been updated:
```
podman run -it ubuntu:22.04 bash

apt-get update && apt-get install -y sudo curl gpg
echo 'deb http://download.opensuse.org/repositories/devel:/tools:/criu/xUbuntu_22.04/ /' | tee /etc/apt/sources.list.d/devel:tools:criu.list
curl -fsSL https://download.opensuse.org/repositories/devel:tools:criu/xUbuntu_22.04/Release.key | gpg --dearmor | tee /etc/apt/trusted.gpg.d/devel_tools_criu.gpg > /dev/null
apt update
apt install -y criu
criu --version
```

**Comment by edsantiago** [comment]
Sigh. [nope](https://api.cirrus-ci.com/v1/artifact/task/4913843032817664/html/int-podman-ubuntu-2204-root-host.log.html). I'll try again tomorrow.

**Comment by edsantiago** [comment]
Failing in [pod create --share-parent test](https://api.cirrus-ci.com/v1/artifact/task/6491108266999808/html/int-podman-ubuntu-2204-root-host.log.html#t--podman-pod-create-share-parent-test--1):
```
Expected
      <string>: host
  to contain substring
      <string>: 1e6d7d50e0d46c60c9906a5518d06969235bf521ea7e71db6dc153131e977e6c
```
The string in question is something to do with cgroups.

@cdoern this is your code, could you spare some cycles to tell me how to fix it? The problem here is that it's failing in cgroupsv1 with runc, because we haven't been testing runc. TIA.

**Comment by edsantiago** [comment]
Failing in [Remote build .containerignore filtering embedded directory](https://api.cirrus-ci.com/v1/artifact/task/4661520918380544/html/int-remote-fedora-36-root-host.log.html#t--Remote-build--containerignore-filtering-embedded-directory-13535---1) with a timeout in the `dd`, even after I switched to `/dev/urandom`.

@jwhonce this is your code, could you please tell me how to fix it? Like, for instance, is it absolutely necessary for the file to be `1G`? Is there something else that could be causing the timeout?

**Comment by edsantiago** [comment]
Sigh, another one. @containers/podman-maintainers who owns the gitlab tests? They're [failing hard](https://api.cirrus-ci.com/v1/task/6691036612263936/logs/setup.log):
```
Obtaining necessary gitlab-runner testing bits
+ ssh some12100dude@localhost -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o CheckHostIP=no env GOPATH=/var/tmp/go go get -u github.com/jstemmer/go-junit-report  # /var/tmp/go/src/github.com/containers/podman/./contrib/cirrus/setup_environment.sh:353 in main()
Warning: Permanently added 'localhost' (ED25519) to the list of known hosts.
go: go.mod file not found in current directory or any parent directory.
	'go get' is no longer supported outside a module.
	To build and install a command, use 'go install' with a version,
	like 'go install example.com/cmd@latest'
	For more information, see https://golang.org/doc/go-get-install-deprecation
	or run 'go help get' or 'go help install'.
```
...and I bet a nickel it's because of the switch go `go 1.18`. This is not something I'm going to try to solve right now, sorry.

**Comment by edsantiago** [comment]
Another new one having to do with checkpoint/restore and pods, I think? [log](https://api.cirrus-ci.com/v1/artifact/task/5879373692665856/html/int-remote-ubuntu-2204-root-host.log.html#t--podman-checkpoint-and-restore-container-out-of-and-into-pod--ipc-net-uts-pid---1). Symptom:
```
 # podman-remote [options] container restore --pod POD-SHA -i /tmp/checkpoint-bar.tar.gz
  Error: cannot restore pod container without --pod
```

**Comment by edsantiago** [comment]
And another hard failure, [timed out waiting for port XXXX](https://api.cirrus-ci.com/v1/artifact/task/5879373692665856/html/int-remote-ubuntu-2204-root-host.log.html#t--authenticated-push--1), ubuntu remote only:
```
Expected
               <*fmt.wrapError | 0xc000a35440>: {
                   msg: "error running \"podman-registry\": : `podman-registry -i docker-archive:/tmp/quay.io-libpod-registry-2.6.tar start` failed: Getting image source signatures\nCopying blob sha256:9d08b7a37338462dc618dd3c25c2c4f713fe3e833f75561950a92bd9130f77ac\nCopying blob sha256:d91b7ec2cc52de561af295d6ea89c55c0c891f8728e65d794e96422f9e815ca7\nCopying blob sha256:849b4a0a6bf5273a3ea90c036c62439d058946f28bb4ad4a8d19255d6237f475\nCopying blob sha256:0341148c78bcafac0ba6f9e0b9056f91af7a380ac1b67ca8d770bdd4eb532bb2\nCopying blob sha256:7444ea29e45e927abea1f923bf24cac20deaddea603c4bb1c7f2f5819773d453\nCopying config sha256:10b45af23ff36baa99dda944a461425494a4bd103f3d4361d30e929f13aa8dda\nWriting manifest to image destination\nStoring signatures\npodman-registry: Timed out waiting for port 5353\n  (exit status 1)",
                   err: <*errors.errorString | 0xc0008971c0>{
                       s: "`podman-registry -i docker-archive:/tmp/quay.io-libpod-registry-2.6.tar start` failed: Getting image source signatures\nCopying blob sha256:9d08b7a37338462dc618dd3c25c2c4f713fe3e833f75561950a92bd9130f77ac\nCopying blob sha256:d91b7ec2cc52de561af295d6ea89c55c0c891f8728e65d794e96422f9e815ca7\nCopying blob sha256:849b4a0a6bf5273a3ea90c036c62439d058946f28bb4ad4a8d19255d6237f475\nCopying blob sha256:0341148c78bcafac0ba6f9e0b9056f91af7a380ac1b67ca8d770bdd4eb532bb2\nCopying blob sha256:7444ea29e45e927abea1f923bf24cac20deaddea603c4bb1c7f2f5819773d453\nCopying config sha256:10b45af23ff36baa99dda944a461425494a4bd103f3d4361d30e929f13aa8dda\nWriting manifest to image destination\nStoring signatures\npodman-registry: Timed out waiting for port 5353\n  (exit status 1)",
                   },
               }
```
Seems networking-related. Are netavark et al up-to-date on Ubuntu?

**Comment by edsantiago** [comment]
Another sigh. [containerized failed again](https://api.cirrus-ci.com/v1/task/6216834272722944/logs/main.log), with checkpoint hangs as expected. But I instrumented the timeout code so it would dump stdout/stderr... and nothing.

**Comment by edsantiago** [comment]
Good news is, `int non-remote ubuntu` tests all pass (root and rootless), indicating that we have a working `criu` again.

**Comment by Luap99** [comment]
> Sigh, another one. @containers/podman-maintainers who owns the gitlab tests? They're [failing hard](https://api.cirrus-ci.com/v1/task/6691036612263936/logs/setup.log):
> 
> ```
> Obtaining necessary gitlab-runner testing bits
> + ssh some12100dude@localhost -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o CheckHostIP=no env GOPATH=/var/tmp/go go get -u github.com/jstemmer/go-junit-report  # /var/tmp/go/src/github.com/containers/podman/./contrib/cirrus/setup_environment.sh:353 in main()
> Warning: Permanently added 'localhost' (ED25519) to the list of known hosts.
> go: go.mod file not found in current directory or any parent directory.
> 	'go get' is no longer supported outside a module.
> 	To build and install a command, use 'go install' with a version,
> 	like 'go install example.com/cmd@latest'
> 	For more information, see https://golang.org/doc/go-get-install-deprecation
> 	or run 'go help get' or 'go help install'.
> ```
> 
> ...and I bet a nickel it's because of the switch go `go 1.18`. This is not something I'm going to try to solve right now, sorry.

I don't know anything about the gitlab test but go 1.18 uses module mode by default and without go.mod this fails. The correct way to install things is `go install` not `go get -u`

**Comment by Luap99** [comment]
@edsantiago This also pins the version to prevent breaking PRs when the tool is updated:
```diff
diff --git a/contrib/cirrus/setup_environment.sh b/contrib/cirrus/setup_environment.sh
index 4952f8dd2..e2d0c655a 100755
--- a/contrib/cirrus/setup_environment.sh
+++ b/contrib/cirrus/setup_environment.sh
@@ -351,7 +351,7 @@ case "$TEST_FLAVOR" in
         slug="gitlab.com/gitlab-org/gitlab-runner"
         helper_fqin="registry.gitlab.com/gitlab-org/gitlab-runner/gitlab-runner-helper:x86_64-latest-pwsh"
         ssh="ssh $ROOTLESS_USER@localhost -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o CheckHostIP=no env GOPATH=$GOPATH"
-        showrun $ssh go get -u github.com/jstemmer/go-junit-report
+        showrun $ssh go install github.com/jstemmer/go-junit-report/v2@v2.0.0
         showrun $ssh git clone https://$slug $GOPATH/src/$slug
         showrun $ssh make -C $GOPATH/src/$slug development_setup
         showrun $ssh bash -c "'cd $GOPATH/src/$slug && GOPATH=$GOPATH go get .'"
```

Although I guess it would make more sense to directly install this into the VM images to reduce downloads and compile time at runtime.

**Comment by Luap99** [comment]
> And another hard failure, [timed out waiting for port XXXX](https://api.cirrus-ci.com/v1/artifact/task/5879373692665856/html/int-remote-ubuntu-2204-root-host.log.html#t--authenticated-push--1), ubuntu remote only:
> 
> ```
> Expected
>                <*fmt.wrapError | 0xc000a35440>: {
>                    msg: "error running \"podman-registry\": : `podman-registry -i docker-archive:/tmp/quay.io-libpod-registry-2.6.tar start` failed: Getting image source signatures\nCopying blob sha256:9d08b7a37338462dc618dd3c25c2c4f713fe3e833f75561950a92bd9130f77ac\nCopying blob sha256:d91b7ec2cc52de561af295d6ea89c55c0c891f8728e65d794e96422f9e815ca7\nCopying blob sha256:849b4a0a6bf5273a3ea90c036c62439d058946f28bb4ad4a8d19255d6237f475\nCopying blob sha256:0341148c78bcafac0ba6f9e0b9056f91af7a380ac1b67ca8d770bdd4eb532bb2\nCopying blob sha256:7444ea29e45e927abea1f923bf24cac20deaddea603c4bb1c7f2f5819773d453\nCopying config sha256:10b45af23ff36baa99dda944a461425494a4bd103f3d4361d30e929f13aa8dda\nWriting manifest to image destination\nStoring signatures\npodman-registry: Timed out waiting for port 5353\n  (exit status 1)",
>                    err: <*errors.errorString | 0xc0008971c0>{
>                        s: "`podman-registry -i docker-archive:/tmp/quay.io-libpod-registry-2.6.tar start` failed: Getting image source signatures\nCopying blob sha256:9d08b7a37338462dc618dd3c25c2c4f713fe3e833f75561950a92bd9130f77ac\nCopying blob sha256:d91b7ec2cc52de561af295d6ea89c55c0c891f8728e65d794e96422f9e815ca7\nCopying blob sha256:849b4a0a6bf5273a3ea90c036c62439d058946f28bb4ad4a8d19255d6237f475\nCopying blob sha256:0341148c78bcafac0ba6f9e0b9056f91af7a380ac1b67ca8d770bdd4eb532bb2\nCopying blob sha256:7444ea29e45e927abea1f923bf24cac20deaddea603c4bb1c7f2f5819773d453\nCopying config sha256:10b45af23ff36baa99dda944a461425494a4bd103f3d4361d30e929f13aa8dda\nWriting manifest to image destination\nStoring signatures\npodman-registry: Timed out waiting for port 5353\n  (exit status 1)",
>                    },
>                }
> ```
> 
> Seems networking-related. Are netavark et al up-to-date on Ubuntu?

Likely still some parallel cni and netavark use?

**Comment by edsantiago** [comment]
>Likely still some parallel cni and netavark use?

Well, the failure is 100% and it's been happening since my first iteration on this PR. That is: I've never seen it pass. That suggests something harder than a race condition.

**Comment by Luap99** [comment]
> > Likely still some parallel cni and netavark use?
> 
> Well, the failure is 100% and it's been happening since my first iteration on this PR. That is: I've never seen it pass. That suggests something harder than a race condition.

I am not saying it is a race, just that something is still using CNI while the e2e test use netavark, check the iptables output to be sure.

**Comment by Luap99** [comment]
> 
>     2. Can some Go expert please fix this stupid code so it actually emits useful output? https://github.com/containers/podman/blob/b5612df55060317bcef92de48e5b2cd400374814/test/utils/utils.go#L367-L372
> 
> 
> Right now that fails with "command timed out but I'm not going to tell you what command it was nor what its output was". I think it might be slightly more helpful to say "this was the command that timed out, and this was the output I got up to this point". I can't figure it out from the GInkgo docs, and have spent much too long on it already.
> 
> Thank you!

2. https://github.com/containers/podman/pull/15021 should fix the command output.

**Comment by edsantiago** [comment]
OK, we need to get rid of `registry.redhat` from our tests. It is flaking way too much.

**Comment by cevich** [comment]
@edsantiago FWIW, [these images passed in buildah CI.](https://github.com/containers/buildah/pull/4074).  I haven't opened PRs in any of the others.  Most don't depend on Ubuntu (so the new version here won't be of much consequence).  When CI passes here, will you help me update the other repos?

> OK, we need to get rid of registry.redhat from our tests. It is flaking way too much.

I've seen this recently as well.  However, IIRC the `ubi` image is used in a lot of different places :S  I'm not certain how locked to that image the tests are or if they could use something else like alpine from our libpod quay namespace.

> [`int podman fedora-36 root container`](https://cirrus-ci.com/task/4603560670265344)

I didn't look at the underlying errors, but I question if those tests should be running as part of the podman-in-podman tests?

**Comment by cevich** [comment]
The `bud podman fedora-36 root host` [failure is a flake.](https://cirrus-ci.com/task/6433148018884608)

**Comment by cevich** [comment]
@edsantiago in case it's a useful comparison, the [Cirrus: Use pre-installed bats](https://github.com/containers/podman/pull/14719) PR has these same images but a smaller diff.  Besides any flakes, that might be a useful point for comparison for any script/code changes here.

**Comment by edsantiago** [comment]
@cevich that PR cannot merge as-is. It can merge with older VM images, or it can wait for this 2204 PR, but not both. The runc changes are enormous (and still in progress) and you need those in your PR. I suggest just waiting for this to merge.

**Comment by cevich** [comment]
> I suggest just waiting for this to merge.

Yep, that's what I was intending.  It's using the same images as this, for extra testing of non-Ubuntuj VMs w/o the changes here.  That was my "comparison" suggestion/point.

File: .cirrus.yml
Comment (cevich) [inline .cirrus.yml:181]: @edsantiago I don't remember checking this or not, so maybe a duplicate comment:  In `lib.sh` there is a list of "pass-through" env. var. names.  Please make sure this one is on that list to ensure the variable and value are passed into (for example) rootless and podman-in-podman CI scripts.

File: contrib/cirrus/setup_environment.sh
Comment (cevich) [inline contrib/cirrus/setup_environment.sh:90]: Yes, I think it is.  Please feel free to nuke it.

File: contrib/cirrus/setup_environment.sh
Comment (cevich) [inline contrib/cirrus/setup_environment.sh:353]: A comment explaining why this is necessary would be helpful.  Our setup for running the gitlab unit tests was carefully crafted based on explicit instructions from the gitlab runner maintainers.  I don't recall this version-reference being in them.  But in general, our task should simply follow the "latest and greatest" upstream.  If we deviate and it breaks, it can be time-consuming to engage with them (they have other priorities).

File: .cirrus.yml
Comment (edsantiago) [reply .cirrus.yml:181]: That's a really good point, thanks. It looks like the test [passed in int-rootless](https://api.cirrus-ci.com/v1/artifact/task/5577721127370752/html/int-podman-fedora-36-rootless-host.log.html#t--Podman-info--check-runtime--1), but not in containerized, so I have work to do. I've also added a triple-check so the test will fail if `$CIRRUS_CI` is defined but `$CI_DESIRED_RUNTIME` is not.

File: contrib/cirrus/setup_environment.sh
Comment (Luap99) [reply contrib/cirrus/setup_environment.sh:353]: go get simply does not work with go 1.18, you have to go install.

You could add `@latest` to always use the latest but IMO this will just start breaking on a random PR, we really have to stop doing this and only update versions via PRs. I am tired of having things break on my PRs because of external changes.

Comment (edsantiago) [reply contrib/cirrus/setup_environment.sh:353]: Furthermore: I cannot find anywhere in that file where it says "this is carefully crafted from anything". If this is fragile code, it must include comments saying so and linking to exact instructions (e.g. on gitlab.com).

File: .cirrus.yml
Comment (cevich) [reply .cirrus.yml:181]: Oof, one thing to watch out for here is: I don't think we set `CIRRUS_CI=true` or `CI=true` in `hack/get_ci_env.sh`.  However that script automatically runs `setup_environment.sh` for developers before handing them a shell.  So checks against those vars could cause problems.

BTW, have you tried `hack/get_ci_vm.sh` recently?  For a number of months,  it has been updated to not require `sudo`, only runs rootless, and only touches files under your `~/.config/gcloud` directory.  All things you took issue with in the past.  Anyway...in case it's useful for you.  It's certainly faster for debugging CI changes than waiting for CIrrus to run them.

Comment (edsantiago) [reply .cirrus.yml:181]: I did try `get_ci_vm.sh` 2-3 weeks ago, rootless (thank you!), but it failed with a long complicated error that I couldn't figure out. I tried a reset(?) arg, still failed. Gave up.

It's 100% OK if `$CIRRUS_DESIRED_RUNTIME` is set without `$CIRRUS_CI`. What is not OK is the converse. And I think (fingers crossed) that this iteration, the one running in CI just now, handles all the corner cases. We'll see in two hours.

File: contrib/cirrus/setup_environment.sh
Comment (cevich) [reply contrib/cirrus/setup_environment.sh:347]: This is not 'go installing' for podman, it's doing so for the upstream gitlab tests.

> I cannot find anywhere in that file where it says

Hrmmm, I thought I had left a comment about that...Yeah, it's this one:
`# Ref: https://gitlab.com/gitlab-org/gitlab-runner/-/issues/27270#note_499585550`

but you're right, it's not clear at all how specialized this setup is.  I s'pose was unintentionally relying on the sheer volume of **setup** section steps in that link. I'll post a docs PR to make it more clear.

Comment (cevich) [reply contrib/cirrus/setup_environment.sh:347]: IMHO, I think Paul's idea is probably okay, so:

`showrun $ssh go install github.com/jstemmer/go-junit-report/v2@latest`

This tool is only needed at the very end of testing, and AFAIK only used to check and convert the results into a useful (for upstream developers) format.

Comment (edsantiago) [reply contrib/cirrus/setup_environment.sh:347]: @Luap99's idea is to use a strict version, not `@latest`, and I ferociously agree. I despise using `latest` because, as Paul, states, it always leads to surprise failures at the least convenient time.

File: .cirrus.yml
Comment (cevich) [reply .cirrus.yml:181]: >  failed with a long complicated error

That may have been the SNAFU w/ AWS.  Should be fixed now.

> I tried a reset(?) arg, still failed. Gave up.

You can try completely removing `~/.config/gcloud` worst-case.  Everythnig in there is generated automatically.  All you'll have to do is re-init (do the browser URL copy-paste thing, select region, etc.).

File: contrib/cirrus/setup_environment.sh
Comment (cevich) [reply contrib/cirrus/setup_environment.sh:347]: Normally yes, but this is a different context.  And there's always the `allow_failures: ...` un-comment option since these aren't "our" tests anyway.  That said, we can roll with the fixed version, but if it breaks, somebody will need to work with the gitlab-runner upstream to fix it (same as for any other gitlab test failure I s'pose).

File: test/e2e/build_test.go
Comment (cevich) [inline test/e2e/build_test.go:557]: I can never remember which is the fast pseudo-random one?  We should use that.  The entropy-pool in these VMs is a problem we've struggled with in different contexts before.

File: test/e2e/info_test.go
Comment (cevich) [inline test/e2e/info_test.go:161]: Would it make sense to just define `CI_DESIRED_RUNTIME` in the Makefile?  That could avoid all this checking-complexity.  CI always runs through the Makefile, and any developer running locally should be doing so as well.

File: test/e2e/build_test.go
Comment (edsantiago) [reply test/e2e/build_test.go:557]: You can safely assume that if I changed it, with a corresponding acknowledgment in the commit message, it is the correct change. But you can also confirm at `random(4)`:
>When read, the /dev/urandom device returns random bytes using a pseudo-random  number generator seeded from the entropy pool.  Reads from this device do not block (i.e., the CPU is not yielded), but  can  incur  an appreciable delay when requesting large amounts of data.

I'm guessing that we're hitting this delay, I just don't know what to do about it, so I'm leaving that as SEP (#15014).

File: test/e2e/info_test.go
Comment (edsantiago) [reply test/e2e/info_test.go:161]: Absolutely not. The Makefile has no knowledge of which VMs want crun and which want runc. Only `.cirrus.yml` should know that.

Although these tests add some duplication, the reason is that I never, ever, _ever_ want to go through this nightmare week again. Nor do I want anyone else to.

Comment (cevich) [reply test/e2e/info_test.go:161]: Relating to `get_ci_vm.sh`: `$CIRRUS_CI` will be unset, but `$CI_DESIRED_RUNTIME`  will be set.

In any case, I'm confused by the mismatch between the condition and skip-message.  In other words, the message mentions one var. but the actual condition is for a different var.

File: test/e2e/manifest_test.go
Comment (cevich) [inline test/e2e/manifest_test.go:298]: @lsm5 was fighting with this one.  Not sure if he opened an issue or not.  IIRC, paul suggested checking the networking and/or firewall.

Comment (edsantiago) [reply test/e2e/manifest_test.go:298]: Yeah. I don't care right now. My only goal is to get this merged. Someone else will have to deal with it.

**Review by cevich — APPROVED** [review]
LGTM assuming the tests can pass :confounded:

File: test/e2e/info_test.go
Comment (edsantiago) [reply test/e2e/info_test.go:161]:
```
if $CI_DESIRED_RUNTIME is not set
    if $CIRRUS_CI is not set
        skip, clearly indicating that this is OK
```

File: test/e2e/manifest_test.go
Comment (cevich) [reply test/e2e/manifest_test.go:298]: I smell a massive month-long "FIXME" & bug-fix effort in the team's future :disappointed:

Comment (edsantiago) [reply test/e2e/manifest_test.go:298]: Yep. Absolutely. And I'm 100000% fine with that: small little chunky fixes over time. The alternative is to hold this PR until every single runc bug is fixed, which may take a month or more, and in the meantime more and more runc-related bugs will sneak in.

**Comment by edsantiago** [comment]
@containers/podman-maintainers PTAL and treat as urgent. CI is still running, but I'm dearly hoping that the tenth time is the charm.

I am truly sorry for the disgusting size of this PR. There is no choice. To recap the problem: we haven't been testing with runc, and a lot of don't-work-with-runc tests have snuck in during that window. Some are easy fixes, others, I've filed issues for. I've also had to file issues for not-obviously-related-to-runc problems that are cropping up. It is my considered belief that it will be much easier to fix those gradually than for this PR to wait for them to be solved.

I cannot be around for most of Friday, but will check in early morning and then midafternoon.

**Comment by edsantiago** [comment]
Whew - CI is finally green.

**Comment by edsantiago** [comment]
...and, re-pushing because #14876 merged and nobody's going to look at this PR until tomorrow anyway.

**Comment by cdoern** [comment]
> ...and, re-pushing because #14876 merged and nobody's going to look at this PR until tomorrow anyway.

Getting my test writing skills ready for whatever 14867 introduced... I think I avoided most of the issues.

**Comment by edsantiago** [comment]
@cdoern your PR includes lots of "bps" strings, so I'm guessing all these new [bps failures](https://api.cirrus-ci.com/v1/artifact/task/5640505697501184/html/int-podman-ubuntu-2204-root-host.log.html) are related. I'm outta here for the day (LONG DAY) but would appreciate your help. TIA.

**Comment by cdoern** [comment]
> @cdoern your PR includes lots of "bps" strings, so I'm guessing all these new [bps failures](https://api.cirrus-ci.com/v1/artifact/task/5640505697501184/html/int-podman-ubuntu-2204-root-host.log.html) are related. I'm outta here for the day (LONG DAY) but would appreciate your help. TIA.

sorry about that, the good news is these failures are cropping up with both iops and bps devices so I can narrow it down. will keep you updated

**Comment by edsantiago** [comment]
Don't sweat it. I really want to get this merged, because I don't want to be here on Friday, so I've resubmitted with `Skips` which you can then fix at your leisure.

**Review by vrothberg — COMMENTED** [review]
LGTM

Restarted the flake.  Thanks everybody!

**Comment by edsantiago** [comment]
Thanks everyone.

There may be in-flight PRs that breaks cgroups V1, such that once they merge they will break CI for everyone. Please consider that during your PR reviews over the next week, and please suggest that everyone rebase.

**Comment by cevich** [comment]
Thanks @edsantiago for your efforts here, so much fun isn't it :wink: Seriously though, I'm happy to see you successfully navigated building new VM images and integrating them into podman CI.  I know there's some testing fallout from this merging, however I don't believe I got an answer to my update question:  Could you help me update these VM images in c/skopeo, c/image, and c/storage?  [I've got one going for buildah already.](https://github.com/containers/buildah/pull/4074)

**Comment by edsantiago** [comment]
@cevich I'm sorry, I seem to have missed that question, and can't even find it in my email archives. If I understand your reference correctly, you would like me to submit `.cirrus.yml` PRs for those three projects, updating the `cXXX` number? If so, would you like me to wait until https://github.com/containers/automation_images/pull/157 merges instead? 

If I misunderstood, could you please point me at your original request, or help me understand? Again, I'm sorry for not being able to find it.

File: test/e2e/build_test.go
Comment (cevich) [reply test/e2e/build_test.go:557]: In case anybody does some archaeology on this change:  GCE uses QEMU/KVM VMs and does not provide a virtual RNG for them (we asked).  That means the entropy pool is incredibly limited, and we rely on the (pre-enabled) `rngd` service to refill it.  In other words, tests should never rely on fast good random numbers, only fast less-good ones or VERY slow good ones.

**Comment by cevich** [comment]
> submit .cirrus.yml PRs for those three projects, updating the cXXX number? If so, would you like me to wait until https://github.com/containers/automation_images/pull/157 merges instead?

Yes please.  There's no need to wait on 157, it doesn't change anything in the VM images, only a few tooling containers.  There are 14-repos total that use GCE images, and probably 8 desperately in need of an update. Most of the time once podman CI is passing, it goes pretty smoothly, though the Ubuntu version and runc update could cause some hiccups.  So help with this is very much appreciated.  

- aardvark-dns  (**Good enough**)
- buildah  (**DONE**)
- BuildSourceImage
- common
- conmon
- dnsname
- image (I got it)
- netavark  (**Good enough**)
- oci-seccomp-bpf-hook
- podman  (**DONE**)
- podman-py (I got it)
- skopeo  (I got it)
- storage (I got it)
- udica (I got it)

> If I misunderstood, could you please point me at your original request, or help me understand? Again, I'm sorry for not being able to find it.

No worries, comments get eaten by github sometimes, and it's not like we receive a small amount of github mail.
Found it (it was burried): https://github.com/containers/podman/pull/14972#issuecomment-1191521107

**Comment by edsantiago** [comment]
Well, we need to rebuild VMs anyway (#15025), so I'll wait until that's fixed.

I tried Re-run on your PR, it still fails (same criu bug on ubuntu), so no point in doing anything until that's fixed.

**Comment by cevich** [comment]
Ugh, -sigh-, okay, I guess I thought it would be simple this time :cry: Okay, I'll /hold all the PR's I already opened.

**Pull request merged**
